subsys/net/lib/dns/Kconfig

```diff
@@ -172,6 +173,16 @@ config MDNS_RESPONDER_DNS_SD_SERVICE_TYPE_ENUMERATION
 	  performs DNS-SD Service Type Enumeration according to RFC 6763,
 	  Chapter 9. By doing so, Zephyr network services are discoverable
 	  using e.g. 'avahi-browse -t -r _services._dns-sd._udp.local'.
+
+config MDNS_RESPONDER_DNS_SD_ANNOUNCE
+	bool "DNS Service Discovery announcements"
+	help
+	  This option will, in addition to responding, announce
+	  MDNS services when new IP addresses are added.
+
+	  Note: Make sure the NET_MGMT_EVENT_STACK_SIZE is large enough to
+	  send announcements.
```

Comment:
I realize that you are just changing existing code and probably do not want to tweak this but I think we should refactor this part of the code to be more robust. Existing event handler callback calls various functions but as its stack is quite limited, it would be more preferrable to move extra processing in the callback to a separate k_work. This way user does not need to figure out the mgmt event stack size at all when things crash weirdly.
We had the same issue in ethernet.c where the carrier on/off handling was originally in event callback but it was moved to a separate k_work that does the actual processing of the event. I suggest we do something similar here.